bsip-0084.md

```diff
@@ -0,0 +1,195 @@
+
+
+# Specifications
+
+## 1. New asset flag "voting_allowed"
+
+A new asset flag/permission "voting_allowed" will be introduced. At the time of the hardfork, all existing UIAs will have the corresponding permission set. Future assets can have the permission set only if they are UIAs not MPAs nor PMs.
+
+As usual with flags, the flag can be changed only if the permission is set. The permission can be unset any time, but can be set only when supply is zero.
+
+The flag must not be used before the time of the hardfork.
+
+
+## 2. New operation "create_elected_authority"
+
+**Note 1:** Since the overall computational overhead for this voting mechanism is significant, the height of the fee should reflect this.
+
+**Note 2:** The new asset flag `voting_allowed` is only checked for this operation. If the flag is removed from an asset, any existing elected authorities are unaffected.
+
+**Fields:**
+
+* `account_id_type creator` - the account to pay the fee, also the only one who can delete the authority
+* `asset_id_type voting_asset` - the asset on which to base the voting
+* `unsigned_int num_members` - the number of authority members to vote on if fixed, or 0 otherwise
+* `unsigned_int min_members` - the minimum number of authority members to elect, or 0 if fixed
+* `unsigned_int max_members` - the maximum number of authority members that can be elected, or 0 if fixed
+* `flat_set<account_id_type> candidates` - a list of candidates eligible for voting, or empty
+* `optional<asset_id_type> candidate_holders` - an asset that candidates must hold to be eligible for voting
+* `bool proxy_allowed` - indicates if proxy voting is allowed
+* `optional<time_point_sec> vote_until` - an optional ending date after which voting slates are frozen
+
+**Validation:**
+
+The operation must not be used before the time of the hardfork.
+
+* `creator` must exist, must have lifetime membership, and must have sufficient balance to pay the fee.
+* `voting_asset` must exist and must have the `voting_allowed` flag set.
+* If `num_members` is 0 then `min_members` and `max_members` must both be positive.
+* If `num_members` is positive then both `min_members` and `max_members` must equal 0.
+* If `candidates` is not empty then
+    * Its size must be greater or equal to `max(num_members,min_members)`.
+    * `candidate_holders` must not be present.
+* If `candidate_holders` is present the `candidates` must be empty.
+* `vote_until`, if present, must be in the future
+
+**Evaluation:**
+
+A new object type `elected_authority_object` is introduced. The operation creates such an object using the fields from the operation.
+
+A new, empty authority is created. The desired number of members for the authority is set to `max(num_members,min_members)`.
+
+Half of the operation fee is set aside and stored in the `elected_authority_object`.
```

Comment:
What will this half be used? Do we need new operations to deposit to it and/or withdraw from it?

Reply:
It will be returned upon deletion.
Elected authorities are expensive for the chain. This mechanism is meant as an incentive to actually delete elected authorities that are no longer in use.

Reply:
I guessed it. Please add the rationale to the document.

Reply:
Done.